pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -343,39 +399,308 @@ func (r *ReconcileVirtualMachineImport) addWatchForImportPod(instance *v2vv1.Vir
 	)
 }
 
-func (r *ReconcileVirtualMachineImport) importDisks(provider provider.Provider, instance *v2vv1.VirtualMachineImport, mapper provider.Mapper, vmName types.NamespacedName) error {
+// convertGuest starts a Job to run virt-v2v on the target VM
+func (r *ReconcileVirtualMachineImport) convertGuest(provider provider.Provider, instance *v2vv1.VirtualMachineImport, vmName types.NamespacedName) (bool, error) {
+	// find the vmspec
+	vmSpec := &kubevirtv1.VirtualMachine{}
+	err := r.client.Get(context.TODO(), vmName, vmSpec)
+	if err != nil {
+		return false, err
+	}
+
+	configMap, err := r.findLibvirtDomainConfigMap(vmName)
+	if err != nil {
+		return false, err
+	}
+	if configMap == nil {
+		configMap, err = r.makeLibvirtDomainConfigMap(instance, vmSpec)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	job, err := r.findGuestConversionJob(vmName)
+	if err != nil {
+		return false, err
+	}
+	// the job doesn't exist, so create it
+	if job == nil {
+		job = r.makeGuestConversionJobSpec(vmSpec)
+		// Set VirtualMachineImport instance as the owner and controller so it'll be cleaned up
+		// when the instance is removed.
+		if err := controllerutil.SetControllerReference(instance, job, r.scheme); err != nil {
+			return false, err
+		}
+		err := r.client.Create(context.TODO(), job)
+		if err != nil {
+			return false, err
+		}
+
+		processingCond := conditions.NewProcessingCondition(string(v2vv1.ConvertingGuest), "Running virt-v2v", corev1.ConditionTrue)
```

Comment:
for debug purpose, would it be useful to include the v2v job name as part of the message for this condition "Running virt-v2v job-vm-name..." ?
which will allow the user that monitors the CR to get the job namespaced-name quickly ?